controller/customdomain_utils.go

```diff
@@ -104,6 +113,83 @@ func FindCustomDomainCondition(conditions []customdomainv1alpha1.CustomDomainCon
 	return nil
 }
 
+// Take an ingress controller managed by the custom domains operator and release it back to the
+// cluster ingress operator. Also schedule it onto customer worker nodes from the Red Hat managed infra
+// nodes.
+func (r *CustomDomainReconciler) returnIngressToClusterIngressOperator(reqLogger logr.Logger, instance *customdomainv1alpha1.CustomDomain) (ctrl.Result, error) {
+	reqLogger.Info(fmt.Sprintf("Removing operator management labels from %s's underlying ingress controller", instance.Name))
+
+	ingressName := instance.Name
+	customIngress := &operatorv1.IngressController{}
+
+	reqLogger.Info(fmt.Sprintf("Fetching ingress controller: %s/%s", ingressOperatorNamespace, ingressName))
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: ingressOperatorNamespace,
+		Name:      ingressName,
+	}, customIngress)
+
+	if err != nil {
+		if kerr.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			reqLogger.Error(err, fmt.Sprintf("Ingresscontroller %s in %s namespace not found", ingressName, ingressOperatorNamespace))
+			return reconcile.Result{}, nil
```

Comment:
Curious if we should have CDO recreate the ingresscontroller and pass it off to the ingress operator in this case?

Reply:
It's a line ball, but I think we probably shouldn't, given that if a customer explicitly deletes an IC I would expect they'd think that it would not come back.